### PR TITLE
Add PHP 8.0 support

### LINKED ISSUE #9: PHP 8.0 support
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes

#### Summary

To be prepared for the december release of PHP 8.0, this repository has some additional TODOs to be tested against the new major version.

In order to make this repository compatible, one has to follow these steps:

- [ ] Modify `composer.json` to provide support for PHP 8.0 by adding the constraint `~8.0.0`
- [ ] Modify `composer.json` to drop support for PHP less than 7.3
- [ ] Modify `composer.json` to implement phpunit 9.3 which supports PHP 7.3+
- [ ] Modify `.travis.yml` to ignore platform requirements when installing composer dependencies (simply add `--ignore-platform-reqs` to `COMPOSER_ARGS` env variable)
- [ ] Modify `.travis.yml` to add PHP 8.0 to the matrix (*NOTE*: Do not allow failures as PHP 8.0 has a feature freeze since 2020-08-04!)
- [ ] Modify source code in case there are incompatibilities with PHP 8.0

### PR DESCRIPTION
**Pull request by bfoosness**
|    Q          |   A
|-------------- | ------
| New Feature   | yes

### Description

Close #9

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,38 +12,26 @@ env:
...
-    - php: 7.2
+    - php: 7.4
       env:
         - DEPS=lowest
-    - php: 7.2
+    - php: 7.4
       env:
         - DEPS=latest
-    - php: 7.3
+    - php: 'nightly'
```
Comment (snapshotpl) [inline .travis.yml]: `8.0` is now supported

File: .travis.yml [hunk 2]
```diff
@@ -12,38 +12,26 @@ env:
...
       env:
         - DEPS=lowest
-    - php: 7.3
+        - COMPOSER_ARGS="--no-interaction --ignore-platform-reqs"
```
Comment (snapshotpl) [inline .travis.yml]: If `laminas/laminas-coding-standard` goes to `2.2` then probably we don't need this ignore

Comment (bfoosness) [reply .travis.yml]: I was able to install dependencies on PHP 8 just fine even though [1.0.0](https://github.com/laminas/laminas-json/blob/185ec9ffaf10427da0f3ad3fd42d8646ff934a9c/composer.json#L32) is required. I'll remove this now.

File: .travis.yml [hunk 3]
```diff
@@ -12,36 +12,22 @@ env:
...
-    - php: 7.1
+    - php: 7.3
       env:
         - DEPS=lowest
-    - php: 7.1
+    - php: 7.3
       env:
         - DEPS=latest
-        - CS_CHECK=true
-        - TEST_COVERAGE=true
```
Comment (froschdesign) [inline .travis.yml]: These tests cannot be removed. Please add again. Thanks!

Comment (bfoosness) [reply .travis.yml]: Should they be added to each version block? That was unclear to me given they were only present on the 7.1 version previously.

Comment (froschdesign) [reply .travis.yml]: Sorry, only for one version. You can use version 7.4.

You can compare with other components. For example: https://github.com/laminas/laminas-hydrator/blob/4.1.x/.travis.yml

**Comment by bfoosness** [comment]
@froschdesign, is there anything else blocking this PR? Thanks in advance!

**Review by froschdesign — APPROVED** [review]
LGTM 👍

**Comment by froschdesign** [comment]
@bfoosness 
Unfortunately yes, because the [automatic release process](https://github.com/laminas/automatic-releases) is missing for this component.

**Comment by bfoosness** [comment]
@froschdesign, I'm happy to create a new PR based on https://github.com/laminas/laminas-server/commit/943a743b2e2b687278c867c31c27e00fe6d84cff if that will help.

**Review by Slamdunk — APPROVED** [review]

**Pull request merged**